src/util/GuardedStream.ts

```diff
@@ -35,8 +35,10 @@ export function isGuarded<T extends NodeJS.EventEmitter>(stream: T): stream is G
  * See https://github.com/solid/community-server/pull/462#issuecomment-758013492 .
  */
 function guardingErrorListener(this: Guarded, error: Error): void {
-  // Only fall back to this if no other listeners are attached
-  if (this.listenerCount('error') === 1) {
+  // Only fall back to this if no new listeners are attached since guarding started.
+  // Not storing the index when guarding starts since listeners could be removed.
+  const idx = this.listeners('error').indexOf(guardingErrorListener);
+  if (idx === this.listenerCount('error') - 1) {
```

Comment:
Actually, there’s no need to look up the actual index. We just need to check if the last listener is this one.

Reply:
Plus, also no need to count separately. Just store the error listeners and check the last one.

Reply:
> Actually, there’s no need to look up the actual index. We just need to check if the last listener is this one.

Yea I should just request the array once. I blame the fact that this is attempt version 4 which caused things to shift several times. 😄 

> Plus, also no need to count separately. Just store the error listeners and check the last one.

Not sure what you mean with this though.

Reply:
> Not sure what you mean with this though.

Indeed, looking up the array once and just using its length.

Reply:
c05933f

And yes I just see now that there is a typo in that commit message damnit.